app/scripts/models/account.js

```diff
@@ -55,6 +55,7 @@ define(function (require, exports, module) {
     // move forward, very hard to move back.
     sessionTokenContext: undefined,
     uid: undefined,
+    verificationMethod: undefined,
```

Comment:
@seanmonstar, @vershwal - @rfk and I had a chat about this today and @rfk suggested something slightly heretical - what if we ignore this case for the sake of progressing the rest of the PR? Instead of worrying about this immediately, log how often users sign in using a passwordless login and an unconfirmed account, open an issue to track that number, and decide in the future whether it warrants further effort. If the user sees a "confirm your email" screen, it'll allow them to "resend" to get a verification email, won't it?